### ISSUE TITLE
[purs ide] Module local definitions lose documentation after fast-rebuild

### ISSUE THREAD

**Issue by kritzcreek** [purescript/purescript#3858]
## Description

Observed in https://github.com/nwolverson/vscode-ide-purescript/issues/142:

When a fast rebuild is triggered and the `rebuildCache`<a href="#note1" id="note1ref"><sup>[1]</sup></a> is populated  the documentation isn't preserved properly in that process.

## To Reproduce

With a module like:
```purescript
module A where

-- | MyType's documentation
type MyType = Int
```

1. Start IDE server
2. See tooltips with docs for `MyType`
3. Rebuild
4. Tooltip without docs
5. Change to a different module and rebuild it
6. Return to the original module and tooltips have docs again

## Expected behavior

Documentation tooltips should be preserved through the rebuild cache

## PureScript version

0.13.6

<a id="note1" href="#note1ref"><sup>[1]</sup></a> After running a quick rebuild for a module we store an Externs for it that exports every member to give completions and type information for module private identifiers

**Comment by hdgarrood** [comment]
> After running a quick rebuild for a module we store an Externs for it that exports every member to give completions and type information for module private identifiers

Does this mean that after doing a fast-rebuild on a particular module, you can import private members from it in other modules downstream of it?

**Comment by kritzcreek** [comment]
> Does this mean that after doing a fast-rebuild on a particular module, you can import private members from it in other modules downstream of it?

No, the generated externs information is solely held in memory in the `ide` process. https://github.com/purescript/purescript/blob/63e90fe46323a5cdc2b1362ca857867499e1eff0/src/Language/PureScript/Ide/Types.hs#L210
